File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/product/dtos/ProductDTO.java
package vn.com.ltdt.Coffee_Shop.product.dtos;

import jakarta.validation.constraints.NotNull;

import java.util.List;

public record ProductDTO(
        int id,
        @NotNull(message = "The name cannot be null")
        String name,
        @NotNull(message = "The description cannot be null")
        String description,
        double rate,
        long rateSum,
        @NotNull(message = "The category Id cannot be null")
        int categoryId,
        String categoryName,
        boolean isActive,
        @NotNull(message = "The details cannot be null")
        List<ProductDetailDTO> details,
        List<String> images
) {
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/remote/ProductDTO.kt
package com.ltdt.coffeeshop_android_native.data.remote

import com.ltdt.coffeeshop_android_native.data.domains.Product

class ProductDTO(
    val id: Int,
    val name: String,
    val description: String,
    val rate: Double,
    val rateSum: Long,
    val categoryId: Int,
    val categoryName: String,
    val isActive: Boolean,
    val details: List<ProductDetailDTO>,
    val images: List<String>
)

fun ProductDTO.toEntity() = Product(
    id = id,
    name = name,
    description = description,
    rate = rate,
    categoryId = categoryId,
    categoryName = categoryName,
    isActive = isActive,
    details = details.map { it.toEntity() },
    images = images,
    rateSum = rateSum
)